### PR TITLE
Rewrite of Avatar command.

### PR DESCRIPTION
**Pull request by fernando4578**

### CODE REVIEW

**Comment by metehus** [comment]
corno

File: commands/avatar.js [hunk 1]
```diff
@@ -1,35 +1,32 @@
 exports.run = function( bot, message, args, helper, config, Rules ) {
```
Comment (metehus) [inline commands/avatar.js]: why all these blank spaces?

File: commands/avatar.js [hunk 2]
```diff
@@ -1,35 +1,32 @@
...
-
-	message.channel.send( embed );
+    Rules.apply({ needArg: false, onlyOwner: false, needAttch: false, needPermission: "EMBED_LINKS" });
+
+    if ( typeof Rules.fine() === "string" ) return message.reply( Rules.fine() );
+    
+    function buildEmbed( avatar, author, title, color ) {
+        console.log( author );
+        embed.setAuthor(author, avatar);
+        embed.setTitle(title + author);
```
Comment (metehus) [inline commands/avatar.js]: you could just use
```
embed.setAuthor(author, avatar)
  .setTitle(title + author)
  ...
```

File: commands/avatar.js [hunk 3]
```diff
@@ -1,35 +1,32 @@
...
+        embed.setColor(color);
+        embed.setImage(avatar);
+    }
+
+    let mention = message.mentions.member.first(),
```
Comment (metehus) [inline commands/avatar.js]: why using `let` insted of `const` if you not changing their values

File: commands/avatar.js [hunk 4]
```diff
@@ -1,35 +1,32 @@
...
-	}
-
-	message.channel.send( embed );
+    Rules.apply({ needArg: false, onlyOwner: false, needAttch: false, needPermission: "EMBED_LINKS" });
+
+    if ( typeof Rules.fine() === "string" ) return message.reply( Rules.fine() );
+    
+    function buildEmbed( avatar, author, title, color ) {
+        console.log( author );
+        embed.setAuthor(author, avatar);
```
Comment (metehus) [inline commands/avatar.js]: you could use embed as a function parameter

File: commands/avatar.js [hunk 5]
```diff
@@ -1,35 +1,32 @@
...
+    let mention = message.mentions.member.first(),
+        member = mention ? mention : message.member,
+        { user } = member,
+        USTools = new helper.USTools(),
+        embed = helper.zerinhoEmbed( member );
+        
+    if ( args[ 0 ] === 'server' ) buildEmbed( USTools.displayGuildIcon( message.guild ), message.guild.name, "Icone de ", "RANDOM" );
+    
+    else buildEmbed(
+        user.displayAvatarURL(null, 2048),
```
Comment (metehus) [inline commands/avatar.js]: [User.displayAvatarURL](https://discord.js.org/#/docs/main/master/class/User?scrollTo=displayAvatarURL) needs to have an object as parameter, use this instead:
`user.displayAvatarURL({ size: 2048 })`

File: commands/avatar.js [hunk 6]
```diff
@@ -1,35 +1,32 @@
...
+        mention ? user.name : '',
+        mention ? "Avatar de " : "Seu avatar.",
```
Comment (metehus) [inline commands/avatar.js]: add a period as in "Seu avatar.":
```
mention ? `Avatar de ${mentionUser}.` : "Seu avatar.",
```
or idk

File: commands/avatar.js [hunk 7]
```diff
@@ -1,35 +1,32 @@
...
-			embed.setTitle( titles[ i ] + authors[ i ] );
-			embed.setImage( results[ i ] );
-			embed.setColor( colors[ i ] );
-
-			i = 3;
-		}
-	}
-
-	message.channel.send( embed );
+    Rules.apply({ needArg: false, onlyOwner: false, needAttch: false, needPermission: "EMBED_LINKS" });
```
Comment (metehus) [inline commands/avatar.js]: why the fuck you have to give`onlyOwner` and `needAttach` as false on every fucking command? why dont you put these parameters as already false?

Comment (zerinho6) [reply commands/avatar.js]: That's what i noticed!....one day after uploading every command with that...i will do something about it on the next update

File: commands/avatar.js (hunk 1 above)
Comment (zerinho6) [reply commands/avatar.js]: That's not a real problem.

File: commands/avatar.js (hunk 3 above)
Comment (metehus) [inline commands/avatar.js]: ``message.mentions.first().member``

Comment (zerinho6) [inline commands/avatar.js]: That'll also throw a error if no one was mentioned.

File: commands/avatar.js [hunk 8]
```diff
@@ -1,35 +1,32 @@
...
+
+    if ( typeof Rules.fine() === "string" ) return message.reply( Rules.fine() );
+    
+    function buildEmbed( avatar, author, title, color ) {
+        console.log( author );
```
Comment (zerinho6) [inline commands/avatar.js]: Get rid of this.

File: commands/avatar.js (hunk 3 above)
Comment (metehus) [reply commands/avatar.js]: but later (line 26)it checks if theres a mention, if not, it uses the message author:
```
 mention ? user.name : ''
```

Comment (zerinho6) [reply commands/avatar.js]: Yeah but it'll throw a error before it even reaches that line, so it won't work, i tested.

File: commands/avatar.js (hunk 7 above)
Comment (zerinho6) [reply commands/avatar.js]: Done, now he'll need to change that too.
```suggestion
    Rules.apply({ needPermission: "EMBED_LINKS" });
```

**Comment by zerinho6** [comment]
Seriusly, codacy have a problem?

**Pull request merged**